internal/provider/common_utils.go

```diff
@@ -0,0 +1,52 @@
+package provider
```

Comment:
We could use a `tar` package of something ? (`tar.File`  ?). `common_utils.go` doesn't really say anything.

Reply:
Refactored this